### DISCUSSION TITLE
Superposition of Velocity Deficits

### DISCUSSION THREAD

**Discussion by seltzera** [NatLabRockies/floris#817] [Q&A]
I wish to use the wake superposition proposed by Niayifar and Porté-Agel in 2016 in FLORIS. Has this already been implemented by anyone in the community, or can someone help me on which steps to take to add it on my own?

I have taken a look into how the other superposition models are coded, e.g. the "sosfs.py" in the wake_combination folder, but it did not give me much to work with in terms of adding a new form of superposition. 

The velocity deficit superposition principle I refer to is:
<img width="288" alt="Screenshot 2024-02-22 at 13 07 31" src="https://github.com/NREL/floris/assets/72613423/a4163ffe-95a1-4edb-863c-b01df4bd8a03">

Where _U_i_ is the velocity at the turbine _i_, and _U_ki_ is the wake velocity of the turbine _k_ at turbine _i_ considering only those turbines whose wakes interact with turbine _i_. 

(From Niayifar, A. and Port ́e-Agel, F. (2016). Analytical modeling of wind farms: A new approach for power prediction. Energies, 9:741.)

**Comment by misi9170** [comment]
Hi @seltzera,

I'm looking back through the discussions realized that somehow we missed this. If you are still interested, I believe that the superposition model that you are interested in is actually already implemented in FLORIS. We refer to this model as the "freestream linear superposition" model, which can be used by changing the `wake: model_strings: combination_model` field on the FLORIS input file to `fls` (in most example input files, the default is the `sosfs` model, or "sum of squares freestreem superposition" model).

Regarding the more general question about adding new superposition models: To do this, you'll want to add your model to the floris/core/wake_combination/ folder, following the format of sosfs.py and fls.py. You'll then need to add the model to the `MODEL_MAP` in floris/core/wake.py, alongside `"fls"`, `"sosfs"`, etc. and ensure that you assign your model to the `wake: model_strings: combination_model` field on your FLORIS input file.

(note that those file locations are for FLORIS v4; for FLORIS v3, look for floris/simulation/wake_combination/ folder and floris/simulation/wake.py)
